flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -853,6 +855,9 @@ public static void main(final String[] args) {
                             "",
                             ""); // no prefix for the YARN session
 
+            final CommandLine commandLine = CliFrontendParser.parse(cli.allOptions, args, true);
+            DynamicPropertiesUtil.encodeDynamicProperties(commandLine, flinkConfiguration);
```

Comment:
This also changes the configuration that has already been passed into `FlinkYarnSessionCli`. I'm not entirely sure about the impact. Maybe it's safer to clone the configuration before modifying it.

Reply:
Thanks your review. But i think that clone is not right. Because yarn-session.sh is also support to use "-D <property=value>" to overwrite flink.conf.yaml. The cmd parameter priority is must higher than the configuration file.

Reply:
I noticed that dynamic properties are applied in `FlinkYarnSessionCli#toConfiguration`, which is called in `FlinkYarnSessionCli#run`. That implies `FlinkYarnSessionCli` expects the configuration being passed in does not include dynamic properties.

Obviously, applying the dynamic properties in `FlinkYarnSessionCli#toConfiguration` is too late for the security configurations. I think we have 2 options:
1. Apply the dynamic properties earlier. We would need to carefully check the codes between the original and new applying position, making sure applying the properties earlier would not cause problems. (That's what I meant by "not entirely sure about the impact".)
2. Or we can keep the applying of dynamic properties as is, and make a copy of the configuration that only affects the security settings.

Reply:
Thanks for your review. Yes, you are right. I  have modified as you said "make a copy of the configuration that only affects the security settings".